### PR TITLE
add annotations to configmaps

### PR DESCRIPTION
**Pull request by DMilmont**
This adds annotations to the following configmap objects:

Webserver
StatsD

Motivation:
We use spinnaker to deploy this helm chart, however spinnaker by default [versions configmaps](https://spinnaker.io/docs/guides/user/kubernetes-v2/best-practices/#version-your-configmaps-and-secrets). This forces us to fork this helm chart and modify it to work with spinnaker, by adding annotations to disable spinnaker versioning. This would take us a step closer to not having to do that anymore. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: chart/templates/configmaps/statsd-configmap.yaml [hunk 1]
```diff
@@ -32,6 +32,10 @@ metadata:
 {{- with .Values.labels }}
 {{ toYaml . | indent 4 }}
 {{- end }}
+{{- if .Values.statsd.configMapAnnotations }}
+  annotations:
+{{- toYaml .Values.statsd.configMapAnnotations | nindent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/statsd-configmap.yaml]:
```suggestion
{{- with .Values.statsd.configMapAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
{{- end }}
```

File: chart/templates/configmaps/webserver-configmap.yaml [hunk 2]
```diff
@@ -32,6 +32,10 @@ metadata:
 {{- with .Values.labels }}
 {{ toYaml . | indent 4 }}
 {{- end }}
+{{- if .Values.webserver.configMapAnnotations }}
+  annotations:
+{{- toYaml .Values.webserver.configMapAnnotations | nindent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/webserver-configmap.yaml]:
```suggestion
{{- with .Values.webserver.configMapAnnotations }}
  annotations:
    {{- toYaml .| nindent 4 }}
{{- end }}
```

**Review by hussein-awala — CHANGES_REQUESTED** [review]
You need to [install pre-commit](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#pre-commit-hooks) to fix static checks

**Comment by DMilmont** [comment]
@hussein-awala I installed pre-commit to fix static checks and committed changes.

**Comment by potiuk** [comment]
> @hussein-awala I installed pre-commit to fix static checks and committed changes.

There are few more suggestions from @hussein-awala

File: chart/templates/configmaps/statsd-configmap.yaml (hunk 1 above)
Comment (DMilmont) [reply chart/templates/configmaps/statsd-configmap.yaml]: Thanks for the suggestion, I went and updated. In the future it would be nice to provide some context on suggestions like these!

File: chart/templates/configmaps/statsd-configmap.yaml [hunk 3]
```diff
@@ -29,9 +29,13 @@ metadata:
     release: {{ .Release.Name }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     heritage: {{ .Release.Service }}
-    {{- with .Values.labels }}
-      {{- toYaml . | nindent 4 }}
-    {{- end }}
+{{- with .Values.labels }}
+{{ toYaml . | indent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/statsd-configmap.yaml:34]: Why did you change this?
```suggestion
    {{- with .Values.labels }}
      {{- toYaml . | nindent 4 }}
    {{- end }}
```

File: chart/templates/configmaps/statsd-configmap.yaml [hunk 4]
```diff
@@ -29,9 +29,13 @@ metadata:
...
+{{- with .Values.statsd.configMapAnnotations }}
+  annotations:
+{{- toYaml . | nindent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/statsd-configmap.yaml:38]: Add indent
```suggestion
  {{- with .Values.statsd.configMapAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: chart/templates/configmaps/webserver-configmap.yaml [hunk 5]
```diff
@@ -29,9 +29,13 @@ metadata:
     release: {{ .Release.Name }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     heritage: {{ .Release.Service }}
-    {{- with .Values.labels }}
-      {{- toYaml . | nindent 4 }}
-    {{- end }}
+{{- with .Values.labels }}
+{{ toYaml . | indent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/webserver-configmap.yaml:34]: same here:
```suggestion
    {{- with .Values.labels }}
      {{- toYaml . | nindent 4 }}
    {{- end }
```

File: chart/templates/configmaps/webserver-configmap.yaml [hunk 6]
```diff
@@ -29,9 +29,13 @@ metadata:
...
+{{- with .Values.webserver.configMapAnnotations }}
+  annotations:
+{{- toYaml . | nindent 4 }}
+{{- end }}
```
Comment (hussein-awala) [inline chart/templates/configmaps/webserver-configmap.yaml:38]: Add indent
```suggestion
  {{- with .Values.webserver.configMapAnnotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

**Review by hussein-awala — CHANGES_REQUESTED** [review]
I think you had a conflict and you fixed it by overriding some changes from #29917 which improved the format of the templates.

Also you can improve the format of the new annotations by adding some indents as #29917 did.

**Comment by amoghrajesh** [comment]
@hussein-awala I will take this up as we also need some custom labels/annotations onto our configmaps.
I will complete this PR
